statistics/statistics.go

```
@@ -212,94 +249,150 @@ func (c *Column) mergeBuckets(bucketIdx int64) {
 	return
 }
 
-// columnToPB converts Column to ColumnPB.
-func columnToPB(col *Column) (*ColumnPB, error) {
-	data, err := codec.EncodeValue(nil, col.Values...)
+// Table represents statistics for a table.
+type Table struct {
+	Info    *model.TableInfo
+	Columns []*Column
+	Indices []*Column
+	Count   int64 // Total row count in a table.
+	Pseudo  bool
+}
+
+// SaveToStorage saves stats table to storage.
+func (t *Table) SaveToStorage(ctx context.Context) error {
+	_, err := ctx.(sqlexec.SQLExecutor).Execute("begin")
+	txn := ctx.Txn()
+	version := txn.StartTS()
+	SetStatisticsTableCache(t.Info.ID, t, version)
+	deleteSQL := fmt.Sprintf("delete from mysql.stats_meta where table_id = %d", t.Info.ID)
+	_, err = ctx.(sqlexec.SQLExecutor).Execute(deleteSQL)
 	if err != nil {
-		return nil, errors.Trace(err)
+		return errors.Trace(err)
+	}
+	insertSQL := fmt.Sprintf("insert into mysql.stats_meta (version, table_id, count) values (%d, %d, %d)", version, t.Info.ID, t.Count)
+	_, err = ctx.(sqlexec.SQLExecutor).Execute(insertSQL)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	deleteSQL = fmt.Sprintf("delete from mysql.stats_columns where table_id = %d", t.Info.ID)
+	_, err = ctx.(sqlexec.SQLExecutor).Execute(deleteSQL)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	deleteSQL = fmt.Sprintf("delete from mysql.stats_buckets where table_id = %d", t.Info.ID)
+	_, err = ctx.(sqlexec.SQLExecutor).Execute(deleteSQL)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, col := range t.Columns {
+		col.saveToStorage(ctx, t.Info.ID, false)
```

Why ignore the error?